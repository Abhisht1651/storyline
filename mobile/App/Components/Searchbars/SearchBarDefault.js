// @flow

import React, { useState, useEffect } from 'react'
import { View, TextInput, TouchableOpacity, Platform } from 'react-native'
import styles from '../Styles/SearchBarStyles/SearchBarDefaultStyle'
import IconSearch from '../../Images/Icons/icon-search-small.svg'
import IconClose from '../../Images/Icons/icon-close-search.svg'

type Props = {
  placeholder: string,
  onSearchHandler: (arg: string) => {}
}

const SearchBarDefault = (props: Props) => {
  const { placeholder, onSearchHandler } = props
  const [crossIconHidden, setCrossIconHidden] = useState(true)
  const [value, setValue] = useState('')

  useEffect(() => {
    onSearchHandler(value)
  }, [value])

  const changeTextHandler = (text: string) => {
    if (text !== '') {
      setCrossIconHidden(false)
      setValue(text)
    } else {
      setCrossIconHidden(true)
      setValue(text)
    }
  }

  const onClearHandler = () => {
    setCrossIconHidden(true)
    setValue('')
  }
  return (
    <View style={styles.container}>
      <View style={styles.searchIcon}>
        <IconSearch height="100%" />
      </View>
      <TextInput
        placeholder={placeholder}
        placeholderTextColor="#8F8FAF"
        style={styles.textinput}
        value={value}
        selectionColor="#7B46E4"
        spellCheck={false}
        autoCorrect={false}
        keyboardType={Platform.OS === 'ios' ? 'default' : 'visible-password'}
        onChangeText={changeTextHandler}
      />
      {!crossIconHidden && (
        <TouchableOpacity style={styles.clear} onPress={onClearHandler}>
          <IconClose height="100%" />
        </TouchableOpacity>
      )}
    </View>
  )
}

SearchBarDefault.defaultProps = {
  placeholder: 'Search'
}

export default SearchBarDefault
