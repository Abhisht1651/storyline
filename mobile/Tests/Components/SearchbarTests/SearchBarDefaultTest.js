import { shallow } from 'enzyme'
import React from 'react'

import { TextInput } from 'react-native'
import SearchBarDefault from '../../../App/Components/Searchbars/SearchBarDefault'

describe('SearchBarDefault', () => {
  it('should match to snapshot', () => {
    const component = shallow(<SearchBarDefault />)
    expect(component).toMatchSnapshot()
  })
  it('SearchBarDefault onChangetText', () => {
    const component = shallow(<SearchBarDefault />)
    component.find(TextInput).prop('onChangeText')('Random')
    expect(component.find(TextInput).prop('value')).toEqual('Random')
  })
})
