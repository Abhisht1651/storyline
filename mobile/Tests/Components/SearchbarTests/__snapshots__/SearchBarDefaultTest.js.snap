// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarDefault should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F8F8",
      "borderRadius": 28.75,
      "flexDirection": "row",
      "height": 57.5,
      "width": 650,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 26.285714285714285,
        "justifyContent": "center",
        "width": 70,
      }
    }
  >
    <SvgMock
      height="100%"
    />
  </View>
  <Component
    allowFontScaling={true}
    autoCorrect={false}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="#8F8FAF"
    rejectResponderTermination={true}
    selectionColor="#7B46E4"
    spellCheck={false}
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Light",
        "fontSize": 27,
        "letterSpacing": -0.11,
        "paddingBottom": 11.5,
        "paddingTop": 11.5,
        "width": "70%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
